lib/brains/views/customer.py

```diff
@@ -22,12 +23,27 @@ def create(request):
         return Response(form.errors, status=400)
 
     uuid = form.cleaned_data['uuid']
+
+    # Create the solitude objects.
+    buyer, buyer_created = Buyer.objects.get_or_create(uuid=uuid)
+    log.info('Buyer {0}: {1}'
+             .format('created' if buyer_created else 'exists', buyer.pk))
+
+    braintree, braintree_buyer_created = (
+        BraintreeBuyer.objects.get_or_create(buyer=buyer))
+    log.info('BraintreeBuyer {0}: {1}'
+             .format('created' if braintree_buyer_created else 'exists',
+                     braintree.pk))
```

Comment:
Its all in a transaction when it fails the whole transaction is rolled back because solitude has transactions from the middleware.